Add Markdown custom field type

MeganerdDev opened the pull request:
<!--
    Thank you for your interest in contributing to Nautobot! Please note
    that our contribution policy recommends that a feature request or bug
    report be opened for approval prior to filing a pull request. This
    helps avoid wasting time and effort on something that we might not
    be able to accept.

    Please indicate the relevant feature request or bug report below.
-->
# Closes: #4006 
# What's Changed
<!--
    Please include:
    - A summary of the proposed changes
    - A sectioned breakdown for larger features under ## subheadings
    - Screenshots, example payloads where relevant:
      - Before/After for bugfixes
      - Using a new feature
-->
- Added new custom field type for Markdown
  - Reused CommentField
  - Reused SmallTextarea widget -> not required for PR
- Apply markdown render to template

# NOTE
- I think still we need to perform `render_markdown()` to test for script injection similar to [custom_fields.py#L588](https://github.com/nautobot/nautobot/blob/452f02852489d60d9620ffa5fd67ab955c2181cd/nautobot/extras/models/customfields.py#L588) since that doesnt sanitize/check the `field.label` data?


# TODO
<!--
    Please feel free to update todos to keep track of your own notes for WIP PRs.
-->
- [x] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

# SCREENSHOTS
![image](https://github.com/nautobot/nautobot/assets/114895043/d16b87a4-bf93-4eab-a3c6-6fcc14e7047d)
![image](https://github.com/nautobot/nautobot/assets/114895043/6dc7301a-84fe-4990-bd49-3558fe5dbd9c)
![image](https://github.com/nautobot/nautobot/assets/114895043/4962a314-69f5-481a-a7f2-853c5b8208b8)

MeganerdDev commented:
Definitely open for feedback!
- I tested this on Nautobot v1.5.22b1 to be working
- I am using the SmallTextarea widget, which could instead just be the default widget

bryanculver commented:
PR might look ugly until #4020 goes in.

glennmatthews commented on nautobot/extras/models/customfields.py:
```diff
@@ -31,6 +31,8 @@
     StaticSelect2,
     StaticSelect2Multiple,
     add_blank_choice,
+    CommentField,
+    SmallTextarea,
 )
```
Nitpick: let's keep the entries here in alphabetical order rather than adding to the end.

glennmatthews reviewed (commented):
Looks both straightforward and useful - my favorite type of PR! :-)

glennmatthews commented:
Oh oops, looks like our CI isn't checking for it in `develop-1.6` yet - could you please add a `changes/4006.added` changelog fragment as part of this PR?

MeganerdDev commented:
> Oh oops, looks like our CI isn't checking for it in `develop-1.6` yet - could you please add a `changes/4006.added` changelog fragment as part of this PR?

Thanks Glenn, you previously reminded me on another pull request on the changelog fragment. I'll try and get that jammed in my brain! Fragment `./changes/4006.added` added
